### PR TITLE
Initial TS-4100 support

### PR DESCRIPTION
**Pull request by ts-kris**
Brings in support for TS-4100 standalone or paired with TS-8551, TS-8100, and TS-8820. 

Also adds the ts_wdt microcontroller watchdog timer driver with updates for 5.10. However, it is currently unused on the TS-4100 in favor of the CPU WDT.

### CODE REVIEW

File: arch/arm/boot/dts/Makefile [hunk 1]
```diff
@@ -640,6 +640,10 @@ dtb-$(CONFIG_SOC_IMX6UL) += \
 	imx6ul-pico-hobbit.dtb \
 	imx6ul-pico-pi.dtb \
 	imx6ul-phytec-segin-ff-rdk-nand.dtb \
+	imx6ul-ts4100.dtb \
+	imx6ul-ts4100-8.dtb \
+	imx6ul-ts4100-7.dtb \
```
Comment (randytsi369) [inline arch/arm/boot/dts/Makefile:644]: Does order matter here? E.g. 
imx6ul-xxxx.dtb
imx6ul-xxxx-7.dtb
imx6ul-xxxx-8.dtb
imx6ul-xxxx-16.dtb
versus how it is now?

**Review by randytsi369 — DISMISSED** [review]
Changes have been reviewed. No issues that I am aware of, but I have not tested these updates.

File: arch/arm/boot/dts/Makefile (hunk 1 above)
Comment (ts-kris) [reply arch/arm/boot/dts/Makefile:644]: Order doesn't matter, that was just a typo it would seem.

**Comment by ts-kris** [comment]
Pushed two small additions. Info print from ts_wdt on register, and swapped dtb ordering in makefile.

**Review by randytsi369 — APPROVED** [review]
Changes have been reviewed (again). No issues that I am aware of, but I have not tested these updates.

**Review by markfeathers — APPROVED** [review]

**Comment by ts-kris** [comment]
Since the next few PRs are quite intertwined, I'm moving to start using merge commits rather than rebasing PRs.

**Pull request merged**